tracing-attributes/tests/support.rs

```diff
@@ -1,3 +1,4 @@
 #[path = "../../tracing/tests/support/mod.rs"]
+#[allow(clippy::module_inception)]
```

Comment:
fwiw, the "module inception" is necessary because the path attribute _requires_ the module to have the same name as the one pointed to by the path, and if we didn't have a module in `tests` wrapping the path attr, every test would need to have the path attribute itself (which is a pain). if you want to add a comment here explaining why we do this...not a blocker though.

Reply:
> take a look at opening a PR for clippy on ci.

I think we should at least consider using https://github.com/actions-rs/clippy-check to run Clippy (rather than just running `cargo clippy` directly) since it has the rather nifty feature of being able to display Cippy lints inline in GitHub's PR interface.